jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/sys/DatabaseOps.java

```diff
@@ -198,6 +199,20 @@ public static void compact(DatasetGraphSwitchable container) {
             LOG.debug(String.format("Compact %s -> %s\n", db1.getFileName(), db2.getFileName()));
 
             compact(container, loc1, loc2);
+
+            if ( shouldDeleteOld ) {
+                Path loc1Path = IOX.asPath(loc1);
+                LOG.info("Deleting old database after successful compaction (old db path='" + loc1Path + "')...");
+
+                try {
```

Comment:
Should use try-with-resources here. (See `Files.walk` `@apiImpl` javadoc.)
This closes the stream which in turn closes the internal `DirectoryStreams`.
```
        try (Stream<Path> walk = Files.walk(Paths.get(...)) ) {
            walk.sorted(Comparator.reverseOrder())
                .map(Path::toFile)
                .forEach(File::delete);
        } catch (IOException ex) {
            IO.exception(ex);
        }
```

Reply:
done